hs_core/views/__init__.py

```diff
@@ -601,7 +602,7 @@ def publish(request, shortkey, *args, **kwargs):
 def set_resource_flag(request, shortkey, *args, **kwargs):
     # only resource owners are allowed to change resource flags
     res, _, user = authorize(request, shortkey, needed_permission=ACTION_TO_AUTHORIZE.SET_RESOURCE_FLAG)
-    t = request.POST['t']
+    t = request.POST.get('t', request.data.get('t'))
```

Comment:
I am wondering if  this ```request.data.get(t')``` will always be evaluated? If so then this will raise AttributeError on ```data``` attribute in the case of UI calling this function.